src/faultlog/unresolved_pels.cpp

```diff
@@ -0,0 +1,223 @@
+    // guarded targets physical path is found in the device tree.
+    // to exit the recusive function return 1 to continue return 0
+    GuardedTarget* guardTarget = reinterpret_cast<GuardedTarget*>(priv);
+    ATTR_PHYS_DEV_PATH_Type phyPath;
+    if (!DT_GET_PROP(ATTR_PHYS_DEV_PATH, target, phyPath))
+    {
+        if (strcmp(phyPath, guardTarget->phyDevPath.c_str()) == 0)
+        {
+            guardTarget->target = target;
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void UnresolvedPELs::populate(sdbusplus::bus::bus& bus,
+                              GuardRecords& guardRecords,
+                              std::vector<uint32_t>& processedEid,
+                              json& jsonNag)
+{
+    log<level::INFO>("UnresolvedPELs::populate");
+    using Objects = std::map<sdbusplus::message::object_path, Interfaces>;
+
+    Objects objects;
+    auto method = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "org.freedesktop.DBus.ObjectManager", "GetManagedObjects");
+    auto reply = bus.call(method);
+    reply.read(objects);
+    for (auto& iter : objects)
+    {
+        bool resolved = true;
+        uint32_t id = 0;
+        for (auto& intf : iter.second)
+        {
+            if (intf.first != "xyz.openbmc_project.Logging.Entry")
+            {
+                continue;
+            }
+            for (auto& propertyMap : intf.second)
+            {
+                if (propertyMap.first == "Id")
+                {
+                    auto idPtr = std::get_if<uint32_t>(&propertyMap.second);
+                    if (idPtr == nullptr)
+                    {
+                        continue;
+                    }
+                    id = *idPtr;
+                }
+                else if (propertyMap.first == "Resolved")
+                {
+                    auto resolvedPtr = std::get_if<bool>(&propertyMap.second);
+                    if (resolvedPtr == nullptr)
+                    {
+                        continue;
+                    }
+                    resolved = *resolvedPtr;
+                }
+            }
+            if (resolved)
+            {
+                continue;
+            }
+
+            // get pel json file
+            std::string pel;
+            auto method2 = bus.new_method_call(
+                "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+                "org.open_power.Logging.PEL", "GetPELJSON");
+            method2.append(id);
+            auto resp2 = bus.call(method2);
+            resp2.read(pel);
+            json pelJson = std::move(json::parse(pel));
+
+            // add cec errorlog
+            json jsonErrorLog = json::object();
+            jsonErrorLog["Callout Section"] =
+                pelJson["Primary SRC"]["Callout Section"];
+            bool deconfigured = false;
+            json& primarySRC = pelJson["Primary SRC"];
+            if (primarySRC.contains("Deconfigured") &&
```

Comment:
put !primarySRC["Deconfigured"].is_null() first and then the contains one. By this way it would be more faster. Reason is NULL check is always quicker than finding something and as it is combined with && operator between the two thus if is_null returns false it won't even go for second condition checking

Reply:
the first condition is to check if the key is present and if present if the value is null so the code is correct i guess